### PR TITLE
feat/groth16

### PR DESCRIPTION
**Pull request by rpalakkal**

### CODE REVIEW

File: circuit/src/scaffold.rs [hunk 1]
```diff
@@ -148,6 +149,7 @@ impl<F: Field, P: JsonRpcClient + Clone, A: AxiomCircuitScaffold<P, F>> AxiomCir
             keccak_call_collector: RefCell::new(Default::default()),
             max_user_outputs: USER_MAX_OUTPUTS,
             max_user_subqueries: USER_MAX_SUBQUERIES,
+            max_groth16_pi: 4,
```
Comment (jonathanpwang) [inline circuit/src/scaffold.rs:152]: nit: make a constant

File: circuit/src/subquery/caller.rs [hunk 2]
```diff
@@ -134,6 +134,25 @@ impl<P: JsonRpcClient, F: Field> SubqueryCaller<P, F> {
         HiLo::from_hi_lo([hi, lo])
     }
 
+    pub fn call<T: FetchSubquery<F>>(
```
Comment (jonathanpwang) [inline circuit/src/subquery/caller.rs:137]: add comment what is different between `call` and `handle_subquery`

File: circuit/src/subquery/caller.rs [hunk 3]
```diff
@@ -154,4 +173,68 @@ impl<P: JsonRpcClient, F: Field> SubqueryCaller<P, F> {
...
+        fe.extend(input.proof_bytes);
+        fe.extend(input.public_inputs);
+        assert_eq!(fe.len(), constants.num_fe_per_input);
+        let zero = ctx.load_witness(F::ZERO);
+        fe.resize_with(constants.max_num_fe_per_input_no_hash, || zero);
+        let to_hash = fe
+            .iter()
+            .map(|v| *v.value())
+            .collect::<Vec<_>>()
+            .chunks(NUM_FE_PER_CHUNK)
```
Comment (jonathanpwang) [inline circuit/src/subquery/caller.rs:194]: this is still hardcoded for max_pi=4?

Comment (jonathanpwang) [reply circuit/src/subquery/caller.rs:194]: or it's just set by the max subquery input length allowed?

Comment (rpalakkal) [reply circuit/src/subquery/caller.rs:194]: NUM_FE_PER_CHUNK is just the max subquery input length allowed (13)

**Review by jonathanpwang — APPROVED** [review]

**Pull request merged**